src/index.js

```diff
@@ -24,27 +24,76 @@ const WATCHING = 'WATCHING'
 const NOT_IGNORED = 'NOT_IGNORED'
 const KICKED = 'KICKED'
 const PAUSED = 'PAUSED'
+const SOCKET_EVENTS = ['connect', 'close', 'error', 'ready'];
 
 module.exports = JackdClient
 
 function JackdClient() {
   this.socket = new Socket()
   this.socket.setEncoding('ascii')
+  this.connected = false
+
+  this.socket.on('ready', () => {
+    this.connected = true
+  })
+
+  this.socket.on('close', () => {
+    this.connected = false
+  })
+}
+
+/**
+ * Useful check for environments where network partitioning is common.
+ * @returns {boolean}
+ */
+JackdClient.prototype.isConnected = function() {
+  return this.connected;
+}
+
+/**
+ * Support event listeners on the underlying socket to support external
+ * reconnection logic.
+ * @param event
+ * @param listener
+ */
+JackdClient.prototype.on = function(event, listener) {
+  if (SOCKET_EVENTS.includes(event)) {
```

Comment:
The way this is written,  JackdClient could never be an eventemitter on it's own. Not sure that is a problem or not.

Reply:
It uses an emitter internally to trigger the callbacks passed in (see the createCommandHandler pattern used throughout). But it wasn't emitting those to consumers as there was no .on() function available to register. 

I originally planned to allow consumers to the internal emitter (which is why I accidentally left in the SOCKET_EVENTS array and check) but I decided that was needless scope creep. And the check for 'if (SOCKET_EVENTS.includes(event)) {' doesn't hurt anything.

Reply:
After going back through the original code, I think that makes sense. Seems simpler to use callbacks in this case than exposing the eventemitter.